src/reactionsystem_conversions.jl

```diff
@@ -556,8 +556,8 @@ function nonlinear_convert_differentials_check(rs::ReactionSystem)
         # If the lhs upper level function is not a differential w.r.t. time.
         # If the contenct of the differential is not a variable (and nothing more).
         # If either of this is a case, throws the warning.
-        if Symbolics._occursin(Symbolics.is_derivative, eq.rhs) ||
-           !Symbolics.istree(eq.lhs) ||
+        if hasnode(Symbolics.is_derivative, eq.rhs) ||
+           !Symbolics.is_derivative(eq.lhs) ||
```

Comment:
Is this really what you want? It is different than the meaning of the old code.

Reply:
Yes, but I figured that this one would be stricter and also better. I.e. the only case where a differential equation is allowed (without additional check) is if the lhs is a pure differential with respect to the iv (and the rhs contain no diffs).

Previously I just checked if the lhs was a function, and if it was, that it correspond to the differential. Now we check if it is a differential operator directly (happy to swap back though)